src/main/java/mate/academy/spring/service/impl/CustomUserDetailsService.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.spring.service.impl;
+
+import mate.academy.spring.model.Role;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String name)
+            throws UsernameNotFoundException {
+        User user = userService.findByEmail(name)
+                .orElseThrow(() -> new UsernameNotFoundException("Can't get user: " + name));
+        org.springframework.security.core.userdetails.User.UserBuilder userBuilder =
```

Comment:
Optional! 
You can replace this package in imports `import org.springframework.security.core.userdetails.User.UserBuilder;`